directory/subdirectory.bzl

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylib module containing rules to create metadata about subdirectories."""
+
+load(":providers.bzl", "DirectoryInfo")
+load(":utils.bzl", "get_relative")
+
+visibility("public")
+
+def _subdirectory_impl(ctx):
+    dir = get_relative(ctx.attr.parent[DirectoryInfo], ctx.attr.path, require_dir = True)
+    return [
+        dir,
+        DefaultInfo(files = dir.transitive_files),
+    ]
+
+subdirectory = rule(
+    implementation = _subdirectory_impl,
+    attrs = {
+        "parent": attr.label(
+            providers = [DirectoryInfo],
+            mandatory = True,
+            doc = "A label corresponding to the parent directory.",
+        ),
+        "path": attr.string(
+            mandatory = True,
+            doc = "A relative path to the parent directory",
```

Comment:
Should this be "A path relative to the parent directory" for clarity? (This is the relative path to the subdir, relative to the parent dir, right?)